Implement Order and Product services with respective models and controllers

JzomitiZumatti opened the pull request:
This commit introduces the 'OrderService' and 'ProductService' classes alongside with their corresponding models 'Order' and 'Product'. The services handle basic CRUD operations for their respective entities. Each service is associated with a controller which exposes the service functionality through RESTful APIs.

JzomitiZumatti committed:
- Implement Order and Product services with respective models and controllers

This commit introduces the 'OrderService' and 'ProductService' classes alongside with their corresponding models 'Order' and 'Product'. The services handle basic CRUD operations for their respective entities. Each service is associated with a controller which exposes the service functionality through RESTful APIs.

ExQA commented on src/main/java/com/example/glovo/model/Order.java:
```diff
@@ -1,4 +1,18 @@
 public class Order {
+    private long id;
+    private String customerName;
+    private String customerNumber;
```
Why use this parameter if you already have a unique **id**?

JzomitiZumatti replied:
I mean phone number, not unique id. My bad with variable naming.

ExQA commented on src/main/java/com/example/glovo/service/OrderService.java:
```diff
@@ -0,0 +1,75 @@
+        Product product = Optional.ofNullable(productService.getProductId(productId))
+                .orElseThrow(() -> new IllegalArgumentException("Product with ID " + orderId + " does not exist"));
+        existingOrder.getProducts().add(product);
+        return existingOrder;
```
maybe batter use like `return  existingOrder.getProducts().add(product);`Less code =)

JzomitiZumatti replied:
Got it, thank you, but this line returns boolean value when I need Order, so can't use your advice here.

ExQA reviewed (approved):
Nice work!

IvashchenkoIvanIgorevich commented on src/main/java/com/example/glovo/contoller/OrderController.java:
```diff
@@ -0,0 +1,54 @@
+@RequestMapping("/orders")
+public class OrderController {
+
+    private OrderService orderService;
```
You have not injected service to the spring context, because currently you have only default constructor. Use annotation @AllArgsConstructor or add constructor with service argument

JzomitiZumatti committed:
Rename customerNumber to customerPhoneNumber

Renamed the variable 'customerNumber' to 'customerPhoneNumber' in both the Order model and OrderService, for better clarity. Additionally, the Lombok @AllArgsConstructor annotation was added to the OrderController, which will automatically generate a constructor with parameters for all fields in the class.

IvashchenkoIvanIgorevich reviewed (approved):
Good work!

shaitanaleksander commented on src/main/java/com/example/glovo/service/OrderService.java:
```diff
@@ -0,0 +1,75 @@
+
+    public Order get(long id) {
+        return Optional.ofNullable(orders.get(id))
+                .orElseThrow(() -> new NoSuchElementException("Order with ID " + id + " does not exist"));
```
optional is not suitable here, in this case, return Optional<Order> or remove optional

JzomitiZumatti committed:
Refactor OrderService to return Optional

The get method of OrderService has been refactored to return an Optional instead of throwing an exception directly. This change allows callers to handle the absence of order in a more flexible way. The corresponding changes have been made in OrderController as well.
